packages/components/src/molecules/WalkThrough/WalkThroughCard.tsx

```diff
@@ -0,0 +1,75 @@
+import {useState} from 'react';
+import {motion, AnimatePresence} from 'framer-motion';
+
+import * as S from './WalkThroughCard.styles';
+import {WalkThroughCardProps} from './types';
+
+const WalkThroughCard: React.FC<WalkThroughCardProps> = ({
+  heading,
+  items,
+  mediaItems,
+  onFinish,
+}) => {
+  const [sliceIndex, setSliceIndex] = useState(0);
+  const onNextClickHandler = () => {
+    if (items?.length - 1 === sliceIndex) {
+      onFinish();
+    } else {
+      setSliceIndex(sliceIndex + 1);
+    }
+  };
+
+  const onBackClickHandler = () => {
+    setSliceIndex(sliceIndex - 1);
+  };
+
+  const mediaItem = mediaItems?.find(el => el.index === sliceIndex);
```

Comment:
Lets wrap this in a `useMemo` hook

Reply:
this is a derived value based on useSate